server/src/test/java/org/opensearch/search/aggregations/metrics/CardinalityAggregatorTests.java

```diff
@@ -199,4 +223,222 @@ private void testAggregation(
     ) throws IOException {
         testCase(aggregationBuilder, query, buildIndex, verify, fieldType);
     }
+
+    public void testDynamicPruningFixedValues() throws IOException {
```

Comment:
can we add a test for exceeding max threshold? https://app.codecov.io/gh/opensearch-project/OpenSearch/pull/13821/blob/server/src/main/java/org/opensearch/search/aggregations/metrics/CardinalityAggregator.java#L175

Reply:
Added